groot-btrfs-release/jobs/groot-btrfs/templates/bin/init-store.erb

```diff
@@ -47,7 +47,24 @@ create_and_mount_filesystem() {
   truncate -s $2 $backing_store_file
 
   mkfs.btrfs -f $backing_store_file
-  mount -t btrfs -o user_subvol_rm_allowed $backing_store_file $1
+
+  # It is possible for the filesystem creation to take some more time
+  # in the background when reaching the mount, causing it to fail.
+  # Simply repeat the attempt. But not too often, i.e. not too long.
+
+  trials=15 ;# 30 seconds, given the sleep period
+  until mount -t btrfs -o user_subvol_rm_allowed $backing_store_file $1
+  do
+      trials=$(expr $trials - 1)
+      if test $trials -eq 0 ; then
+	  echo New filesystem has not become ready within 30 seconds.
```

Comment:
Maybe instead of stating 30 seconds, state the number of trials.

Reply:
Or record the starting time and do a calculation here of the real time it took.

Reply:
Done with commit b8c6bb018cce208f3a26c646fded7a3250b96554
(Calculating `trials * sleep_time`, and added an `about` to indicate that it is not fully precise)